bundles/net.tourbook/src/net/tourbook/ui/tourChart/action/ActionXAxisDistance.java

```diff
@@ -1,46 +1,56 @@
- *   
+ * Copyright (C) 2005, 2020  Wolfgang Schramm and Contributors
+ *
  * This program is free software; you can redistribute it and/or modify it under
- * the terms of the GNU General Public License as published by the Free Software 
+ * the terms of the GNU General Public License as published by the Free Software
  * Foundation version 2 of the License.
- *  
- * This program is distributed in the hope that it will be useful, but WITHOUT 
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS 
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
  * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
- * 
- * You should have received a copy of the GNU General Public License along with 
+ *
+ * You should have received a copy of the GNU General Public License along with
  * this program; if not, write to the Free Software Foundation,
- * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA    
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
  *******************************************************************************/
 package net.tourbook.ui.tourChart.action;
 
 import net.tourbook.Messages;
 import net.tourbook.application.TourbookPlugin;
+import net.tourbook.common.UI;
 import net.tourbook.ui.tourChart.TourChart;
 
 import org.eclipse.jface.action.Action;
 
 public class ActionXAxisDistance extends Action {
 
-	private TourChart	fTourChart;
+   private TourChart fTourChart;
 
-	public ActionXAxisDistance(final TourChart tourChart) {
+   public ActionXAxisDistance(final TourChart tourChart) {
 
-		super(Messages.Tour_Action_show_distance_on_x_axis, AS_RADIO_BUTTON);
+      super(Messages.Tour_Action_show_distance_on_x_axis, AS_RADIO_BUTTON);
 
-		this.fTourChart = tourChart;
+      this.fTourChart = tourChart;
 
-		setToolTipText(Messages.Tour_Action_show_distance_on_x_axis_tooltip);
-		setImageDescriptor(TourbookPlugin.getImageDescriptor(Messages.Image__show_distance_on_x_axis));
-		setDisabledImageDescriptor(TourbookPlugin.getImageDescriptor(Messages.Image__show_distance_on_x_axis_disabled));
+      setToolTipText(Messages.Tour_Action_show_distance_on_x_axis_tooltip);
 
-		setChecked(!tourChart.getTourChartConfig().isShowTimeOnXAxis);
-	}
+      String imagePath = Messages.Image__show_distance_on_x_axis;
+      String disabledImagePath = Messages.Image__show_distance_on_x_axis_disabled;
 
-	@Override
-	public void run() {
-		fTourChart.actionXAxisDistance(isChecked());
-	}
+      if (!UI.UNIT_IS_METRIC) {
+         imagePath = Messages.Image__show_distance_on_x_axis_imperial;
+         disabledImagePath = Messages.Image__show_distance_on_x_axis_imperial_disabled;
+      }
+
+      setImageDescriptor(TourbookPlugin.getImageDescriptor(imagePath));
+      setDisabledImageDescriptor(TourbookPlugin.getImageDescriptor(disabledImagePath));

```

Comment:
It looks like, when the measurement system is changed, then this image is not updated.

`TourChart` has a pref listener `net.tourbook.ui.tourChart.TourChart.addPrefListeners()` which already checks the measurement system, there the action image could be updated, access to the action is `tourAction = _allTourChartActions.get(ACTION_ID_X_AXIS_DISTANCE);`